Godeps/_workspace/src/github.com/tendermint/tendermint/vm/vm.go

```diff
@@ -745,12 +769,21 @@ func (vm *VM) call(caller, callee *Account, code, input []byte, value int64, gas
 				stack.Push(newAccount.Address)
 			}
 
-		case CALL, CALLCODE: // 0xF1, 0xF2
+		case CALL, CALLCODE, DELEGATECALL: // 0xF1, 0xF2, 0xF4
 			if !HasPermission(vm.appState, callee, ptypes.Call) {
 				return nil, ErrPermission{"call"}
 			}
 			gasLimit := stack.Pop64()
-			addr, value := stack.Pop(), stack.Pop64()
+			addr := stack.Pop()
+			// NOTE: for DELEGATECALL value is preserved from the original
+			// caller, as such it is not stored on stack for the address
+			// to be called; for CALL and CALLCODE value is stored on stack
+			// and needs to be overwritten from the given value.
```

Comment:
I found this a bit hard to parse, having read it a few times over. Here is a proposed rewording for clarity, but happy to see an even better version if this doesn't convey proper meaning:

```
			// NOTE: for DELEGATECALL value is preserved from the original
			// caller, as such it is not stored on stack for as an argument for
                        // DELEGATECALL and should not be popped. Instead the previous
                        // caller value is use.
                        // for CALL and CALLCODE value is stored on stack
			// and needs to overwrite the original caller value.
```